app/javascript/gobierto_data/webapp/components/sidebar/SidebarDatasets.vue

```diff
@@ -103,47 +82,52 @@ export default {
+      this.showLess = true;
+      if(this.sortedItems.length) {
+        this.activeDatasetColumns = this.sortedItems.find(({ attributes: { slug } = {} }) => slug === newSlug).attributes.columns || {}
+        this.showToggle = Object.keys(this.activeDatasetColumns).length > this.showMaxKeys
+      }
+    }
+  },
+  computed: {
+    activeDatasetVisibleColumns() {
+      if(this.showLess && Object.keys(this.activeDatasetColumns).length) {
+       return Object.keys(this.activeDatasetColumns).slice(0, this.showMaxKeys).reduce((result, key) => {
+          result[key] = this.activeDatasetColumns[key];
+          return result;
+        }, {});
+      } else {
+        return this.activeDatasetColumns;
+      }
+    }
   },
   methods: {
-    orderDatasets() {
-      //TODO Datasets should be order in filter mixin
-      const allDatasets = this.items.sort(({ attributes: { name: a } = {} }, { attributes: { name: b } = {} }) => a.localeCompare(b));
-
-      //Get slug from route params, we need this when user click in any dataset
-      let { id } = this.$route.params || {}
-
-      let slugRoute = id
-      let slugDataset = null
-
-      const indexToggle = allDatasets.findIndex(({ attributes: { slug } = {} }) => slug === slugRoute)
-      this.toggle = indexToggle
-
-      if (this.toggle === -1) {
-        const { attributes: { slug } = {} } = allDatasets[0] || []
-        slugDataset = slug
+    selectCurrentDataset(selectedDatasetSlug) {
+      if(selectedDatasetSlug === undefined && this.sortedItems.length) {
+        selectedDatasetSlug = this.sortedItems[0].attributes.slug
+      }
+      if(selectedDatasetSlug !== undefined) {
+        let selectedDataset = this.sortedItems.find(({ attributes: { slug } = {} }) => slug === selectedDatasetSlug)
+        let filteredArray = this.sortedItems.filter(({ attributes: { slug } = {} }) => slug !== selectedDatasetSlug)
+        this.listDatasets = [selectedDataset].concat(filteredArray)
+        this.currentDatasetSlug = selectedDatasetSlug
       }
-
-      id = this.toggle === -1 ? slugDataset : slugRoute
-
-      let firstElement = allDatasets.find(({ attributes: { slug } = {} }) => slug === id)
-      let filteredArray = allDatasets.filter(({ attributes: { slug } = {} }) => slug !== id)
-      filteredArray.unshift(firstElement)
-      this.listDatasets = filteredArray
-      this.toggle = 0
-      this.sliceColumns(0)
-    },
-    handleToggle(index) {
-      this.showLess = true
-      this.sliceColumns(index)
-      this.toggle = this.toggle !== index ? index : null;
-    },
-    sliceColumns(index) {
-      const allColumns = this.listDatasets ? Object.keys(this.listDatasets[index].attributes.columns) : []
-      this.lessColumns = allColumns.slice(0, this.showMaxKeys)
     },
-    showKeys() {
-      return Object.keys(this.listDatasets[0].attributes.columns).length > this.showMaxKeys
+    handleToggle(slug) {
+      this.currentDatasetSlug = slug
+    }
+  },
+  filters: {
```

Comment:
It's interesting to use a filter to allow translating strings in the HTML

Reply:
We can provide another method here https://github.com/PopulateTools/gobierto/blob/master/app/javascript/lib/shared/modules/vue-filters.js

where we can reduce the verbosity of `I18n.t('gobierto_data.john.doe.value')` using some kind of vue global function: `$t` for instance, in this way `$t(value)` or `$t(gobierto_data.john.doe.value')`

Reply:
Will, let's create an issue for this.